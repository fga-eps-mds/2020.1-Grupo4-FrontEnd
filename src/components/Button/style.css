.button {
    margin-right: 1rem;
    border-radius: 30px;
    text-align: center;
    box-sizing: border-box;
    cursor: pointer;
    padding: 2px 2px;
  
    transition-duration: 0.5s;
    
    display: inline-block;
    outline: none;
    text-decoration: none;
  
    width: 144px;
    height: 28px;
    background: var(--primary-button);
  
  
    font:  700 18px Overpass;
    color: var(--white);
    border: 0;
  }
  
  .button:hover {
    color: white;
    background: var(--primary-button);
  }
  
  .button:hover {
    background: #3C0966;
  }
  
  @media (max-width: 800px) {
    .button {
      text-align: center;
      margin-left: 3px;
    }
   
  }