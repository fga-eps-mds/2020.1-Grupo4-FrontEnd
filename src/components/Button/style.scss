.button {
  min-width: 135px;
  padding: 0.4rem;
  background-color: var(--primary-button);
  color: white;
  text-transform: capitalize;
  text-align: center;
  font-family: Overpass;
  letter-spacing: 0.5px;
  font-weight: bolder;
  border: 1px solid transparent;
  border-radius: 30px;
  cursor: pointer;
  display: flex;
  justify-content: center;

  &:hover {
    background-color: white;
    color: var(--hover-button);
    border: 1px solid var(--hover-button);
  }

  &:active {
    transform: scale(0.95);
    transition: 0.1s;
  }

  &.inverted {
    background-color: white;
    color: var(--hover-button);
    border: 1px solid var(--hover-button);

    &:hover {
      background-color: var(--primary-button);
      color: white;
      border: none;
    }
  }

  &.small {
    padding: 0.2rem;
    font-size: 16px;
    font-weight: bolder;
  }

  &.color {
    color: var(--login-button);
    border-color: var(--login-button);

    &:hover {
      background-color: var(--login-button);
      color: white;
    }
  }

  &.shadow {
    box-shadow: 0px 4px 4px rgba(51, 51, 51, 0.3), 0px 1px 1px rgba(51, 51, 51, 0.15);
  }

  &:focus {outline:0;}
}