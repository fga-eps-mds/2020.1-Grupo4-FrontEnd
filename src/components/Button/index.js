import React, { Children } from 'react';
import './style.css';


function Button({ children, ...otherProps }) {
    return (
        <button className="button" {...otherProps}>
            {children}
        </button>
    );
}


export default Button;
