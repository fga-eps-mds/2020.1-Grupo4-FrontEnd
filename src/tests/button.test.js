import { shallow } from 'enzyme';
import React from 'react';
import Button from '../components/Button';

it('Expect to render Button component', () => {
  expect(shallow(<Button></Button>).length).toEqual(1)
  expect(shallow(<Button></Button>)).toMatchSnapshot()
})

it('Expect to click button', () => {
  const wrapper = shallow(<Button />);
  wrapper.find('button').simulate('click')
})