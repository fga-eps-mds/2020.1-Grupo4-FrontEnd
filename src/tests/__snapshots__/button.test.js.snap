// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Button component 1`] = `
<button
  className="
        
        
        
        
         button"
/>
`;
